rewrite-maven/src/main/java/org/openrewrite/maven/cleanup/RemoveCompileScope.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven.cleanup;
+
+import org.jetbrains.annotations.NotNull;
+import org.openrewrite.*;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.maven.MavenVisitor;
+import org.openrewrite.xml.RemoveContentVisitor;
+import org.openrewrite.xml.tree.Content;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.*;
+
+public class RemoveCompileScope extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Removes the compile scope";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Removes the compile scope for all the dependencies as it is the default scope,and it is redundant.";
+    }
+
+    @Override
+    public List<SourceFile> visit(List<SourceFile> before, ExecutionContext ctx) {
+        // First we collect all the dependencies under dependencyManagement
+        // We should not remove the compile scope for dependencies declared with a different scope within dependencyManagement
+
+        final Map<String, String> seen = new HashMap<>();
+
+        for (SourceFile entry : before) {
```

Comment:
The scope of a dependency might be managed by a parent pom not present in the project which will never be discovered by this loop. 

Instead look at the `MavenResolutionResult` marker which will be present on any `Xml.Document` which is a maven pom. This marker contains the complete set of requested and resolved dependencies, parent poms, etc.

I don't have an example of the exact sort of lookup which is required here, but [AddManagedDependency](https://github.com/yeikel/rewrite/blob/f102c64f95f0580dd78e8899d28d38344456f958/rewrite-maven/src/main/java/org/openrewrite/maven/AddManagedDependency.java#L169) has an example of usage of this marker which you may find helpful